### DISCUSSION TITLE
Self Hosted - locale

### DISCUSSION THREAD

**Discussion by JCV67** [Budibase/budibase#7153] [Help]
Hello, is there a way to set regional settings in a self hosted Budibase system? I need to set the date fields and pickers in pt_PT format. Is it possible?
Thank you

**Comment by melohagan** [comment]
HI @JCV67 

Currently the locale for date fields and pickers defaults to 'en' and is set within the code.

THere's a couple open related issues for it: 
- https://github.com/Budibase/budibase/issues/3551
- https://github.com/Budibase/budibase/issues/5173

**Reply by JCV67** [reply]
Hi @melohagan 
Thank you for your reply, do you know if there is any chance in the near future for us to be able of changing locales?

**Reply by melohagan** [reply]
To be honest I don't see it on the road map: https://github.com/Budibase/budibase/projects/10
It's probably not high up the priority list.

**Comment by JCV67** [comment]
Thank you @melohagan , If only I could format dates as DD:MM:YYYY , but when I try to do it over a column, using  {{date DD:MM:YYYY}} a only get today's date.

**Reply by melohagan** [reply]
Hey @JCV67 That's because you need to provide a date. E.g. `{{ date Form.DatePickerField "DD:MM:YYYY" }}`

*date* is the name of the helper function. You need to provide both a date value and format string.

**Reply by JCV67** [reply]
how would you do it on a table column?

**Reply by melohagan** [reply]
Could you outline further what your use case is please? E.g. is it for filtering, and what are you trying to filter.

**Reply by JCV67** [reply]
To show dates on a column, like this:
![imagem](https://user-images.githubusercontent.com/104722906/184391066-9f496147-21e9-41fa-ab4f-543aa70e7cb7.png)

**Reply by JCV67** [reply]
but in dd:mm:yyy format

**Reply by melohagan** [reply]
I understand. You can use formula columns in this case!
https://docs.budibase.com/docs/formula

So within the *Data* section, add another column to your table, with type formula. Then your expression would be something like:
`{{ date Data de Necessidade "DD:MM:YYYY" }}`

Then within the *Design* section, you can configure the table columns to only show the pretty formula column that you created:
https://docs.budibase.com/docs/table#configure-columns

**Reply by JCV67** [reply]
Thank you very much, I was trying to do it only by configuring table columns in the design section.
